src/libraries/System.Private.DataContractSerialization/src/System/Xml/XmlBufferReader.cs

```diff
@@ -389,59 +370,35 @@ public int ReadUInt31()
         }
 
         public long ReadInt64()
-        {
-            long lo = (uint)ReadInt32();
-            long hi = (uint)ReadInt32();
-            return (hi << 32) + lo;
-        }
+            => BitConverter.IsLittleEndian ? ReadRawBytes<long>() : BinaryPrimitives.ReverseEndianness(ReadRawBytes<long>());
 
-        public unsafe float ReadSingle()
-        {
-            int offset;
-            byte[] buffer = GetBuffer(ValueHandleLength.Single, out offset);
-            float value;
-            byte* pb = (byte*)&value;
-            DiagnosticUtility.DebugAssert(sizeof(float) == 4, "");
-            pb[0] = buffer[offset + 0];
-            pb[1] = buffer[offset + 1];
-            pb[2] = buffer[offset + 2];
-            pb[3] = buffer[offset + 3];
-            Advance(ValueHandleLength.Single);
-            return value;
-        }
+        public float ReadSingle()
+            => ReadRawBytes<float>();
 
-        public unsafe double ReadDouble()
-        {
-            int offset;
-            byte[] buffer = GetBuffer(ValueHandleLength.Double, out offset);
-            double value;
-            byte* pb = (byte*)&value;
-            DiagnosticUtility.DebugAssert(sizeof(double) == 8, "");
-            pb[0] = buffer[offset + 0];
-            pb[1] = buffer[offset + 1];
-            pb[2] = buffer[offset + 2];
-            pb[3] = buffer[offset + 3];
-            pb[4] = buffer[offset + 4];
-            pb[5] = buffer[offset + 5];
-            pb[6] = buffer[offset + 6];
-            pb[7] = buffer[offset + 7];
-            Advance(ValueHandleLength.Double);
-            return value;
-        }
+        public double ReadDouble()
+            => ReadRawBytes<double>();
 
         public decimal ReadDecimal()
         {
-            byte[] buffer = GetBuffer(ValueHandleLength.Decimal, out int offset);
-            ReadOnlySpan<byte> bytes = buffer.AsSpan(offset, sizeof(decimal));
-            ReadOnlySpan<int> span = stackalloc int[4]
+            if (BitConverter.IsLittleEndian)
             {
-                BinaryPrimitives.ReadInt32LittleEndian(bytes),
-                BinaryPrimitives.ReadInt32LittleEndian(bytes.Slice(4)),
-                BinaryPrimitives.ReadInt32LittleEndian(bytes.Slice(8)),
-                BinaryPrimitives.ReadInt32LittleEndian(bytes.Slice(12))
-            };
-
```

Comment:
The changes to parsing was tested against unit tests before adding back fast path for LittleEndian case.

Without fixing this the method will throw exception and abort when deserializing  decimal from binary xml

Reply:
Fixes #73934

Reply:
Note: even when fixing it will throw different exception for invalid decimal than on net framework.
If it is important to add checking for invalid decimal then msybe it is best to do something similar to net framework here instead (and do it before decimal ctor) https://referencesource.microsoft.com/#System.Runtime.Serialization/System/Xml/XmlBufferReader.cs,469

Reply:
@StephenMolloy I did a separate [commit](
https://github.com/Daniel-Svensson/runtime/commit/0776072f333758ae4836c63c55a9afd2a9210dc9) to add the decimal flag validation from net48 but it was not pushed. If you find that check is important and worth the time I can open up a PR. (It is not important to me)